multimodal/src/autogluon/multimodal/predictor.py

```diff
@@ -1450,6 +1485,21 @@ def evaluate(
         A dictionary with the metric names and their corresponding scores.
         Optionally return a dataframe of prediction results.
         """
+        if label is not None:
+            for metric in metrics:
+                if metric is not None and not isinstance(metric, str):
+                    metric = metric.name
+                if metric is not None and metric.lower() in [
+                    "rmse",
+                    "r2",
+                    "pearsonr",
+                    "spearmanr",
+                ]:
+                    self._problem_type = REGRESSION
+            self._label_column = label
+            self._eval_metric_name = eval_metric
+            self.fit(data, data, init_only=True)
+
```

Comment:
Using the code in init and calling fit here looks like a hack. We need a better design.

Reply:
Yes, I'll remove changes for evaluate and we can work on it in next pr.